### ISSUE TITLE
StokesSpectralCube access

### ISSUE THREAD

**Issue by keflavich** [radio-astro-tools/spectral-cube#750]
To access the components of a `StokesSpectralCube`, you presently need to access a private attribute, for example:

```python
cube_I = cube._stokes_data['I']
```

That's not good!  We should probably implement _at least_ a public access like:
```python
cube_I = cube.stokes_data['I']
```

and better still,
```python
cube_I = cube['I']
```

@Kitchi, what do you think?

A PR would be welcome!

**Comment by Kitchi** [comment]
I've got a PR ready with the first suggestion, but I like the second one a lot more. Both @preshanth and I can work on implementing that for the Stokes cubes.

**Comment by preshanth** [comment]
I agree that we should implement something along the second style of 
`cube_I = cube['I']`. 
This will allow for internal conversion functions between the different stokes say. You have a cube that contains ['I','Q','U','V'] then when a user asks for 
`cube_P=cube['P']` we can perform the stokes conversion internally.

**Comment by keflavich** [comment]
closed by #751
